library/htmlspecialchars.inc.php

```diff
@@ -51,6 +51,22 @@ function errorLogEscape($text)
     return attr($text);
 }
 
+/**
+ * Escape variables that are outputted into csv and spreadsheet files.
+ * See here:
+ * https://www.owasp.org/index.php/CSV_Injection
+ *  Basically ensure entry does not start with characters: = + - @
+ *  (another option is to place a apostrophe in front of these character,
+ *   but removing them completely seems more secure; can revisit this if
+ *   removal breaks anything; could add an option to support - for example
+ *   if need to have negative numbers)
+ *  Also remove all trailing spacing
```

Comment:
interesting vulnerability, been thinking about bringing in leaguecsv package to help with csvs and they've got this, https://csv.thephpleague.com/9.0/interoperability/escape-formula-injection/

Reply:
hi @stephenwaite , Been reading more on this. Looks like escaping these characters with apostrophe or tab character would be best. But seems like there is no perfect solution. Looks like the library you pointed to above defaults to tab character. gonna keep researching this,

Reply:
can only do so much, seems like you're on the right track